cimr/processor/query.py

```diff
@@ -0,0 +1,273 @@
+    xenbase           | Xenopus l. + t. DB  | xenbase:XB-GENE-1001990&species=frog
+    zfin              | Zebrafish DB   | zfin:ZDB-GENE-980526-104&species=zebrafish
+
+    Methods
+    -------
+
+    
+
+    Notes
+    -----
+
+    mygene.info website documentation can be found here:
+    docs.mygene.info/en/latest/index.html
+
+    mygene.info is a website for gene annotation queries using the biothings
+    backend. biothings documentation is here
+    biothingsapi.readthedocs.io/en/latest
+
+    cimr currently only utilizes the batch query function using direct queries
+    to the website api
+
+    """
+
+    FIELDS = {'accession', 'alias', 'ensembl.gene', 'ensembl.protein', 
+              'ensembl.transcript', 'entrezgene', 'flybase', 'go', 'hgnc',
+              'homologene', 'hprd', 'interpro', 'mgi', 'mim', 'mirbase', 
+              'name', 'pdb', 'pfam', 'pharmgkb', 'prosite', 'reagent',
+              'refseq', 'reporter', 'retired', 'rgd', 'summary', 'symbol',
+              'tair', 'unigene', 'uniprot', 'wormbase', 'xenbase', 'zfin'
+              }
+
+
+    def __init__(self, 
+                 genes,
+                 headers = {'content-type':'application/x-www-form-urlencoded'}, 
+                 url = 'https://mygene.info/v3/query', 
+                 species = 9606,
+                 scopes = 'alias+symbol+entrezgene+ensembl.gene+ensembl.transcript',
+                 fields = 'name+symbol+taxid+entrezgene+ensembl+alias+refseq'
+                 ):
+        """Initialize the Querier"""
+        self.genes = genes
+        self.headers = headers
+        self.url = url
+        self.species = species
+        self.scopes = scopes
+        self.fields = fields
+
+
+    def make_string(self):
+        """Take the gene_id column input from the data table and turn it
+        into a string to be used in form_query
+        """
+        if type(self.genes) is list:
+            logging.info(f' converting a gene list into a queriable string.')
+            self.genestring = add_plus(self.genes)
+        else:
+            logging.error(f' is the input a list with gene_id\'s?')
+
+    
+    def check_fields(self):
+        """Check if all items indicated in the fields parameter are legit.
+        The list has been updated based on mygene.info documentation on
+        January 2019.
+        """
+        splitted = subtract_plus(self.fields)
+        for field in splitted:
+            if field not in self.FIELDS:
+                raise ValueError(' %s is not a valid field supported in cimr.' % field)
+            else:
+                splitted.remove(field)
+        self.fields = add_plus(splitted)
+    
+
+    def check_ensembl(self):
+        """Commonly used Ensembl ID may include a version identifier. Check for any
+        Ensembl IDs in the feature list and remove the version identifier.
+        """
+        if any('ENSG' in g for g in self.genes):
```

Comment:
If `self.genes` has many elements and each element has many sub-elements, this implementation could be slow. (Forget my comment if the # of elements is not that large).

Reply:
That's true. I opened a new [issue](https://github.com/greenelab/cimr/issues/15) to look into this in my next PR. Thank you!